Split urls before opening files

Example:

```
Could not open file: www/example/_static/doctools.js?v=9bcbadda
```

Maybe splitting at `?` is correct? Apparently these random strings are used for cache busting.  It can cause some sites to look weird

This is probably very low priority right now but it's impossible to serve some static websites with french accents and other symbols like that

```
[ERROR] Could not open file: www/example/articles/s/i/g/Image%7ESignac_-_La_Calanque_edit.jpg_61a7.html
```

URL need to be decoded like %7E etc.

--- a/src/http/http_parser.cpp
+++ b/src/http/http_parser.cpp
@@ -95,7 +95,17 @@ void HttpParser::parse_request_line()
     }
 
     req_.method = v[0];
-    req_.path = v[1];
+
+    size_t qpos = v[1].find("?");
+    if (qpos != std::string::npos) {
+        req_.path = v[1].substr(0, qpos);
+        req_.query_string = v[1].substr(qpos + 1);
+    }
+    else {
+        req_.path = v[1];
+        req_.query_string = "";
+    }
+
     req_.http_version = http_version_from_string(v[2]);
 
     req_.keep_alive = (req_.http_version == kHttpVersion1_1) ? true : false;

--- a/src/http/http_request.hpp
+++ b/src/http/http_request.hpp
@@ -12,6 +12,7 @@ struct HttpRequest {
 
     std::string method;
     std::string path;
+    std::string query_string;
     HttpVersion http_version;
 
     // Try to keep feature parity with HttpResponse if applicable here.

--- a/tests/http_parser_unittest.cpp
+++ b/tests/http_parser_unittest.cpp
@@ -14,6 +14,20 @@ UTEST(HttpParserTest, BasicRequestLine)
     EXPECT_TRUE(p.request().http_version == kHttpVersion1_0);
 }
 
+UTEST(HttpParserTest, RequestWithQueryString)
+{
+    HttpParser p;
+    std::string req = "GET /search.php?q=foo&user=bar HTTP/1.0\r\n";
+
+    p.append_data(req.data(), req.size());
+    ASSERT_TRUE(p.state() == HttpParser::kParsingHeaders);
+
+    EXPECT_TRUE(p.request().method == "GET");
+    EXPECT_TRUE(p.request().path == "/search.php");
+    EXPECT_TRUE(p.request().query_string == "q=foo&user=bar");
+    EXPECT_TRUE(p.request().http_version == kHttpVersion1_0);
+}
+
 UTEST(HttpParserTest, FullRequestZeroBody)
 {
     HttpParser p;

